src/main/java/com/coreos/jetcd/EtcdMaintenanceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.coreos.jetcd;
+
+import com.coreos.jetcd.api.*;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.stub.StreamObserver;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Implementation of maintenance client
+ */
+public class EtcdMaintenanceImpl implements EtcdMaintenance {
+
+    private MaintenanceGrpc.MaintenanceFutureStub futureStub;
+    private MaintenanceGrpc.MaintenanceStub streamStub;
+
+    public EtcdMaintenanceImpl(MaintenanceGrpc.MaintenanceFutureStub futureStub,
+                               MaintenanceGrpc.MaintenanceStub streamStub) {
+        this.futureStub = futureStub;
+        this.streamStub = streamStub;
+    }
+
+    /**
+     * get all active keyspace alarm
+     *
+     * @return alarm list
+     */
+    @Override
+    public ListenableFuture<AlarmResponse> listAlarms() {
+        AlarmRequest alarmRequest = AlarmRequest.newBuilder()
+                .setAlarm(AlarmType.NONE)
+                .setAction(AlarmRequest.AlarmAction.GET)
+                .setMemberID(0).build();
+        return this.futureStub.alarm(alarmRequest);
+    }
+
+    /**
+     * disarms a given alarm
+     *
+     * @param member the alarm
+     * @return the response result
+     */
+    @Override
+    public ListenableFuture<AlarmResponse> disAlarm(AlarmMember member) {
```

Comment:
We probably should typedef it. We should not directly use the type from protobuf. I know we use them for return values today. But we will get rid of them soon.